python/ray/serve/tests/test_standalone2.py

```diff
@@ -419,6 +419,55 @@ def f():
     ray.shutdown()
 
 
+def test_serve_stream_logs(shutdown_ray):
+    """Test that serve logs show up across different drivers."""
+    import tempfile
+
+    subprocess.check_output(["ray", "stop", "--force"])
+    wait_for_condition(
+        _check_ray_stop,
+        timeout=15,
+    )
+    subprocess.check_output(["ray", "start", "--head"])
+    wait_for_condition(
+        lambda: requests.get("http://localhost:52365/api/ray/version").status_code
+        == 200,
+        timeout=15,
+    )
+    ray.init(address="auto", namespace=SERVE_NAMESPACE)
+
+    file1 = """from ray import serve
+@serve.deployment
+class A:
+    def __call__(self):
+        return "Hello A"
+serve.run(A.bind())"""
+
+    file2 = """from ray import serve
+@serve.deployment
+class B:
+    def __call__(self):
+        return "Hello B"
+serve.run(B.bind())"""
+
+    with tempfile.NamedTemporaryFile() as f1, tempfile.NamedTemporaryFile() as f2:
+        f1.write(file1.encode("utf-8"))
+        f1.seek(0)
+        # Driver 1 (starts Serve controller)
+        output = subprocess.check_output(["python", f1.name], stderr=subprocess.STDOUT)
+        assert "Adding 1 replica to deployment default_A" in output.decode("utf-8")
+
+        f2.write(file2.encode("utf-8"))
+        f2.seek(0)
+        # Driver 2 (reconnects to the same Serve controller)
+        output = subprocess.check_output(["python", f2.name], stderr=subprocess.STDOUT)
+        assert "Adding 1 replica to deployment default_B" in output.decode("utf-8")
+
+    serve.shutdown()
+    ray.shutdown()
+    subprocess.check_output(["ray", "stop", "--force"])
```

Comment:
why are these necessary? shouldn't they be handled by the fixture?

Reply:
Ray needs to be started through the cli, otherwise the serve scripts that are run in the test will start their own ray clusters and the test isn't meaningful. Let me pull out this code into a fixture!